CMakeLists.txt

```diff
@@ -278,6 +285,15 @@ if(USE_MKLDNN)
   set_target_properties(dnnl PROPERTIES CXX_CLANG_TIDY "")  # don't lint 3rdparty dependency
 endif()
 
+if(USE_INTGEMM)
+  message(STATUS "Using intgemm")
+  add_subdirectory(3rdparty/intgemm)
```

Comment:
Do we expect a large number of users to compile with intgemm? If not, why force all users to download intgemm instead of making it optional?

```
FetchContent_Declare(
  intgemm
  GIT_REPOSITORY https://github.com/kpu/intgemm.git
  GIT_TAG        77095a0cd5da8a22c4e7ed77588b527d3ee4ada9
)


FetchContent_GetProperties(catch)
if(NOT catch_POPULATED)
  FetchContent_Populate(catch)
  add_subdirectory(${catch_SOURCE_DIR} ${catch_BINARY_DIR})
endif()
```

Reply:
Added, though I'm a bit worried about polluting the includes since there's no longer a 3rdparty/intgemm prefix to them.

Reply:
The aim is that Sockeye will be able to use `pip` installed `mxnet`.  Given the withdrawal of #18904, there isn't a nice way to add to the `pip` version.  Is download or submodule preferred?  

cc @fhieber

Reply:
Back to fetching for now.